server/routes/userRoutes.js

```diff
@@ -106,18 +93,57 @@ module.exports = (db) => {
     const { host_name } = req.body.params;
     dbHelper
       .getWebsiteIDByHostname(host_name)
-      // .getWebsiteIDByHostname("reddit.com")
-      .then((site) => dbHelper.addWebsiteToBlacklist(userId, site.id))
-      .catch((err) => res.json(err));
+      .then((site) => {
+        if (!site) {
+          // Removes '.com' from end of hostname for DB.
+          const remSuffix = host_name.split(".")[0];
+          // Capitalizes first letter of hostname
+          const name = remSuffix.charAt(0).toUpperCase() + remSuffix.slice(1);
+          const category = null;
+          // Creating the website in the database before it can be added to their blacklist
+          dbHelper.addWebsite(host_name, name, category).then((site) => {
+            dbHelper.addWebsiteToBlacklist(userId, site.id);
```

Comment:
Line 105 and 108 (and all routes in general) need to end in a response being sent back to the client, either a success message or an error message if the db operation failed. Add `.then(blacklist=>res.status(201).json(blacklist))` and a `.catch(err=>res.status(500).json(err)`
201 = created
500 = server error (db error)

I think if you leave out the .status it defaults to a success status so it's important to set the status during a failure.